recbole/config/configurator.py

```diff
@@ -440,21 +442,29 @@ def _set_default_parameters(self):
             "split": {"RS": [0.8, 0.1, 0.1]},
             "order": "RO",
             "group_by": "user",
-            "mode": "full",
+            "mode": {"valid": "full", "test": "full"},
         }
         if not isinstance(self.final_config_dict["eval_args"], dict):
             raise ValueError(
                 f"eval_args:[{self.final_config_dict['eval_args']}] should be a dict."
             )
-        for op_args in default_eval_args:
-            if op_args not in self.final_config_dict["eval_args"]:
-                self.final_config_dict["eval_args"][op_args] = default_eval_args[
-                    op_args
-                ]
 
+        default_eval_args.update(self.final_config_dict["eval_args"])
+
+        mode = default_eval_args['mode']
+        # backward compatible
+        if isinstance(mode, str):
+            default_eval_args['mode'] = {'valid': mode, 'test': mode}
+
+        # in case there is only one key in `mode`, e.g., mode: {'valid': 'uni100'} or mode: {'test': 'full'}
+        if isinstance(mode, dict):
+            default_mode = mode.get('valid', mode.get('test', 'full'))
+            default_eval_args['mode'] = {'valid': mode.get('valid', default_mode), 'test': mode.get('test', default_mode)}
+
+        self.final_config_dict["eval_args"] = default_eval_args
```

Comment:
Hi @Sherry-XLL, thank you for your kind words. I would be very happy if I can make a contribution to such a brilliant library.

Regarding your question, these pieces of code can explain more. Here it supports three scenarios:

1.  The original settings, e.g., `mode: full`. It will be converted into `"mode": {"valid": "full", "test": "full"}` in the code.
2. There is only **ONE** key in the settings, e.g., `mode: {valid: full}` or `mode: {test: full}`. It will be extended to TWO keys with the same `mode`: `"mode": {"valid": "full", "test": "full"}`;
3. There are **TWO** keys in the settings, which is the expected version.

One more thing, we here assume users can only use `valid` or `test` as the key in the settings, otherwise, it will raise a `ValueError`.

Thanks!

Reply:
Hello, @ChenglongMa! 

Thank you for your timely update of the documentation! We are very happy to see your expanded function of the evaluation mode. As for the default `.rst` document (_e.g._, `distributed_training.rst`), we still prefer to keep it in the original `str` format. In other words, the default configuration in the document is still `full` or `uni100` instead of `{"valid": "full", "test": "full"}`, which is simple and clear in most cases. The dictionary configuration is only utilized when users want to use different modes for evaluation and test.

Thank you again for your great contributions to RecBole! If you can adjust the default configuration of the document at a convenient time, we are very happy to merge your code into the main branch.

Reply:
Hi @Sherry-XLL, I've reverted the related docs and unit tests to the original versions. Thanks!